go/vrf/ecvrf.go

```diff
@@ -43,3 +44,37 @@ func NewKey(curve elliptic.Curve, sk []byte) *PrivateKey {
 		d:         new(big.Int).SetBytes(sk),           // Use SK to derive the VRF secret scalar x
 	}
 }
+
+// ECVRFParams holds shared values across ECVRF implementations.
+// ECVRFParams also has generic algorithms that rely on ECVRFAux for specific sub algorithms.
+type ECVRFParams struct {
+	suiteString []byte         // single nonzero octet specifying the ECVRF ciphersuite
+	ec          elliptic.Curve // Elliptic curve defined over F
+	//   G - subgroup of E of large prime order.
+	//   q - prime order of group G, ec.Params().N
+	//   B - generator of group G, ec.Params.{Gx,Gy}
+	n        int  // 2n  - length, in octets, of a field element in F.
```

Comment:
Comment says 2n and not n? If intentional, please elaborate.
nit: Delete full stop at the end of this line to match the rest of the comments.

Reply:
2n is intentional, pull straight from the spec

Reply:
Discussion ongoing below